ArduPlane/quadplane.cpp

```diff
@@ -858,6 +868,62 @@ void QuadPlane::hold_hover(float target_climb_rate)
     run_z_controller();
 }
 
+/*
+  control QACRO mode
+ */
+void QuadPlane::control_qacro(void)
+{
+    if (throttle_wait) {
+        motors->set_desired_spool_state(AP_Motors::DESIRED_GROUND_IDLE);
+        attitude_control->set_throttle_out_unstabilized(0, true, 0);
+    } else {
+        check_attitude_relax();
+
+        motors->set_desired_spool_state(AP_Motors::DESIRED_THROTTLE_UNLIMITED);
+
+        // convert the input to the desired body frame rate
+        // Note that the 90 degree Y rotation for copter mode swaps roll and yaw
+        float roll_in = plane.channel_rudder->norm_input();
+        float pitch_in = plane.channel_pitch->norm_input();
+        float yaw_in = -plane.channel_roll->norm_input();
```

Comment:
i think you need a tail sitter enabled check here, this would be very confusing for other quadplanes.

Reply:
I hadn't thought about this enough; it also ignores/bypasses tailsitter.input_type

I was planning to just add QACRO to the list of "modes to avoid" for normal quadplanes...
Since the copter motors on non-tailsitters act perpendicular to the wings, I doubt that it would be a good idea to try to do Qmode aerobatics.

@tridge Your thoughts on this would be appreciated.

Reply:
I don't think you should cut the other Quadplane's out of the fun. These mini talon quadplanes people are building could probably do some acro, in anycase some people might not want to do aerobatics but are more conformable controlling in that mode.  If you don't add a check here I think you need to add one so it refuses to switch to acro if its not a tailsitter.

Reply:
@IamPete1 I modified the first commit to handle non-tailsitters, and added a commit to disable QACRO for non-tailsitters. Could you test with an appropriate non-tailsitter? If that goes well we can drop the second commit.

Reply:
I tested QACRO with the RF8 Big Stik TiltRotorTri and the controls act as one would expect, though I think more work is required to make it useful.

Reply:
I was just going to have a test, i think if it works the we might as well have it

Reply:
Just tested it with the big stik quadplane, works fine, doesn't fly great but that is a air frame problem not a code one. I think you will also have to add it to the descriptions various things that can change flight modes. All the FLTMODE prams, the rc_option ones also and probably open a issue for mission planner and qgc to add it the there enums.

Reply:
just tried my tvbs, this doesn't abide by  tailsitter.input_type. And i get a big drop when switch to acro, the throttle midpoint is not the same as it is for stabalise. ACRO_THR_MID is the way to fix on copter but we don't seem to have that param.

Reply:
hum, maybe the dropping along way is just my bad setup

Reply:
I didn't think anyone would want multicopter input type for QACRO, but I guess that's wrong :)
The throttle handling is the same as for copter acro: mid-stick is set to get_throttle_hover()
Works well with hover throttle learning on the Stryker_quad

Reply:
yeah, i think my hover throttle is probably wrong, i haven't tried hover throttle learning yet. I only learned on copters, not done and plane flying outside of FBWA and above, need all the help I can get ;).